src/envoy/http/metadata_exchange/config.cc

```diff
@@ -34,22 +35,53 @@ using Common::Wasm::Null::Plugin::removeResponseHeader;
 using Common::Wasm::Null::Plugin::replaceRequestHeader;
 using Common::Wasm::Null::Plugin::replaceResponseHeader;
 
-void PluginContext::onCreate() {
-  // TODO(kuat): serialize metadata in the root context instead
-  // populate and encode node metadata
+void PluginRootContext::onConfigure(
+    std::unique_ptr<WasmData> ABSL_ATTRIBUTE_UNUSED configuration) {
   auto metadata =
       getMetadataValue(Common::Wasm::MetadataType::Node, NodeMetadataKey);
   if (metadata.kind_case() == google::protobuf::Value::kStructValue) {
     std::string metadata_bytes;
-    metadata.struct_value().SerializeToString(&metadata_bytes);
+    google::protobuf::io::StringOutputStream md(&metadata_bytes);
```

Comment:
Is this change in order to make serialization deterministic? Why do we need to make it deterministic?

Reply:
lack of determinism has bitten us many times in the past. It also makes testing easier.

Reply:
Ah yeah, indeed it could make test easier when we try to verify the metadata header value. Thanks!

Reply:
Don't know about that. The real issue is depending on bytes as a hash. This is explicitly prohibited by protobuf, and we simply should not depend on the raw bytes.